packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts

```diff
@@ -306,4 +340,25 @@ export class Schedule extends Resource implements ISchedule {
     const isEmptyPolicy = Object.values(policy).every(value => value === undefined);
     return !isEmptyPolicy ? policy : undefined;
   }
-}
\ No newline at end of file
+
+  private renderFlexibleTimeWindow(
+    flexibleTimeWindowMode?: FlexibleTimeWindowMode, maximumWindowInMinutes?: Duration,
+  ): CfnSchedule.FlexibleTimeWindowProperty {
+    if (!flexibleTimeWindowMode || flexibleTimeWindowMode === FlexibleTimeWindowMode.OFF) {
+      return {
+        mode: FlexibleTimeWindowMode.OFF,
+      };
+    }
+
+    if (!maximumWindowInMinutes) {
+      throw new Error('maximumWindowInMinutes must be provided when flexibleTimeWindowMode is set to FLEXIBLE');
+    }
+    if (maximumWindowInMinutes.toMinutes() < 1 || maximumWindowInMinutes.toMinutes() > 1440) {
+      throw new Error(`maximumWindowInMinutes must be between 1 and 1440, got ${maximumWindowInMinutes.toMinutes()}`);
+    }
+    return {
+      mode: flexibleTimeWindowMode,
+      maximumWindowInMinutes: maximumWindowInMinutes.toMinutes(),
+    };
+  }
```

Comment:
Thanks for the fix!

Sorry, this one might be better.

```suggestion
  private renderFlexibleTimeWindow(
    flexibleTimeWindowMode?: FlexibleTimeWindowMode, maximumWindowInMinutes?: Duration,
  ): CfnSchedule.FlexibleTimeWindowProperty {
    const mode = flexibleTimeWindowMode ?? FlexibleTimeWindowMode.OFF;

    if (mode === FlexibleTimeWindowMode.OFF) {
      return {
        mode,
      };
    }

    if (!maximumWindowInMinutes) {
      throw new Error('maximumWindowInMinutes must be provided when flexibleTimeWindowMode is set to FLEXIBLE');
    }
    if (maximumWindowInMinutes.toMinutes() < 1 || maximumWindowInMinutes.toMinutes() > 1440) {
      throw new Error(`maximumWindowInMinutes must be between 1 and 1440, got ${maximumWindowInMinutes.toMinutes()}`);
    }
    return {
      mode,
      maximumWindowInMinutes: maximumWindowInMinutes.toMinutes(),
    };
  }
```